crates/ironrdp-acceptor/src/finalization.rs

```diff
@@ -223,3 +224,27 @@ fn create_font_map() -> rdp::headers::ShareDataPdu {
         entry_size: 0,
     })
 }
+
+fn decode_share_control(input: &[u8]) -> ConnectorResult<rdp::headers::ShareControlHeader> {
+    let data_request = pdu::decode::<pdu::mcs::SendDataRequest<'_>>(input).map_err(ConnectorError::pdu)?;
+    let share_control = rdp::headers::ShareControlHeader::from_buffer(Cursor::new(data_request.user_data))?;
+    Ok(share_control)
+}
+
+fn decode_font_list(input: &[u8]) -> Result<rdp::finalization_messages::FontPdu, Vec<u8>> {
```

Comment:
I would tend to prefer pushing the allocation to the caller, especially given that we don’t have the ownership over `input`:
```suggestion
fn decode_font_list(input: &[u8]) -> Result<rdp::finalization_messages::FontPdu, ()> {
```

```rust
// Callsite
Err(()) => {
    self.input_events.push(input.to_vec());
    (Written::Nothing, FinalizationState::WaitFontList)
}
```